src/libraries/Microsoft.Extensions.Configuration/tests/ConfigurationTest.cs

```diff
@@ -744,6 +744,44 @@ public void SectionWithValueExists()
             Assert.False(sectionNotExists);
         }
 
+        [Fact]
+        public void SectionGetRequiredSectionSuccess()
+        {
+            // Arrange
+            var dict = new Dictionary<string, string>()
+            {
+                {"Mem1", "Value1"},
+                {"Mem1:KeyInMem1", "ValueInMem1"},
+                {"Mem1:KeyInMem1:Deep1", "ValueDeep1"}
+            };
+            var configurationBuilder = new ConfigurationBuilder();
+            configurationBuilder.AddInMemoryCollection(dict);
+            IConfigurationRoot config = configurationBuilder.Build();
+
+            // Act
+            var sectionExists1 = config.GetRequiredSection("Mem1").Exists();
+            var sectionExists2 = config.GetRequiredSection("Mem1:KeyInMem1").Exists();
+
+            // Assert
+            Assert.True(sectionExists1);
+            Assert.True(sectionExists2);
+        }
+
+        [Fact]
+        public void SectionGetRequiredSectionMissingThrowException()
+        {
+            // Arrange
+            var dict = new Dictionary<string, string>()
+            {
+                {"Mem1", "Value1"},
+            };
+            var configurationBuilder = new ConfigurationBuilder();
+            configurationBuilder.AddInMemoryCollection(dict);
+            IConfigurationRoot config = configurationBuilder.Build();
+
+            Assert.Throws<InvalidOperationException>(() => config.GetRequiredSection("Mem2"));
```

Comment:
You might want to add a test for a subsection as well. This is only testing it for a top-level section on `IConfigurationRoot`. Could try a level or two down.

Reply:
I added a sub section